warehouse/forklift/legacy.py

```diff
@@ -481,44 +482,11 @@ def _verify_url_pypi(url: str, project_name: str, project_normalized_name: str)
     )
 
 
-def _verify_url_with_trusted_publisher(url: str, publisher_url: str) -> bool:
-    """
-    Verify a given URL against a Trusted Publisher URL
-
-    A URL is considered "verified" iff it matches the Trusted Publisher URL
-    such that, when both URLs are normalized:
-    - The scheme component is the same (e.g: both use `https`)
-    - The authority component is the same (e.g.: `github.com`)
-    - The path component is the same, or a sub-path of the Trusted Publisher URL
-      (e.g.: `org/project` and `org/project/issues.html` will pass verification
-      against an `org/project` Trusted Publisher path component)
-    - The path component of the Trusted Publisher URL is not empty
-    Note: We compare the authority component instead of the host component because
-    the authority includes the host, and in practice neither URL should have user
-    nor port information.
-    """
-    publisher_uri = rfc3986.api.uri_reference(publisher_url).normalize()
-    user_uri = rfc3986.api.uri_reference(url).normalize()
-    if publisher_uri.path is None:
-        # Currently no Trusted Publishers have an empty path component,
-        # so we defensively fail verification.
-        return False
-    elif user_uri.path and publisher_uri.path:
-        is_subpath = publisher_uri.path == user_uri.path or user_uri.path.startswith(
-            publisher_uri.path + "/"
-        )
-    else:
-        is_subpath = publisher_uri.path == user_uri.path
-
-    return (
-        publisher_uri.scheme == user_uri.scheme
-        and publisher_uri.authority == user_uri.authority
-        and is_subpath
-    )
-
-
 def _verify_url(
-    url: str, publisher_url: str | None, project_name: str, project_normalized_name: str
+    url: str,
+    publisher: OIDCPublisher | None,
+    project_name: str,
+    project_normalized_name: str,
```

Comment:
I might be missing something, but why do we pass both the project name and its normalized version here? Could we instead produce the normalized name within the function body, to save ourselves a parameter here?

(Apologies in advance if there's a good reason to do this that I've missed!)

Reply:
I couldn't find the logic for normalizing the name, other than a reference to a SQL function called `normalize_pep426_name`.

Reply:
Ah, right 🤦. I suppose we could pass the entire `Project` model in, but that's a lot heavier. This seems fine to me, then.